include/highfive/bits/H5Converter_misc.hpp

```diff
@@ -267,32 +267,126 @@ struct data_converter<boost::numeric::ublas::matrix<T>, void> {
 #endif
 
 #ifdef H5_USE_EIGEN
+//compute size for single Eigen Matrix
+template <typename T, int M, int N>
+inline size_t compute_total_size(const Eigen::Matrix<T,M,N>& matrix)
+{
+    return matrix.rows() * matrix.cols();
+}
+
+//compute size for  std::vector of Eigens
+template <typename T, int M, int N>
+inline size_t compute_total_size(const std::vector<Eigen::Matrix<T,M,N>>& vec)
+{
+    return std::accumulate(vec.begin(), vec.end(), size_t{0u},
+                           [](size_t so_far, const auto& v){return so_far + (size_t)v.rows() * (size_t)v.cols(); });
+}
+
+//compute total row size for std::vector of Eigens
+template <typename T, int M, int N>
+inline size_t compute_total_row_size(const std::vector<Eigen::Matrix<T,M,N>>& vec)
+{
+    return std::accumulate(vec.begin(), vec.end(), size_t{0u},
+                           [](size_t so_far, const auto& v){return so_far + (size_t)v.rows(); });
+}
+
+
 // apply conversion to eigen matrix
 template <typename T, int M, int N>
 struct data_converter<Eigen::Matrix<T, M, N>, void> {
 
     typedef typename Eigen::Matrix<T, M, N> MatrixTMN;
 
     inline data_converter(MatrixTMN&, DataSpace& space)
-        : _dims(space.getDimensions()) {
+        : _dims(space.getDimensions()) {    
```

Comment:
you should use `git-clang-format` to automatically format your changes in the git staging area before commiting